cognite/neat/graph/extractors/_dexpi.py

```diff
@@ -0,0 +1,202 @@
+import xml.etree.ElementTree as ET
+from collections.abc import Iterable
+from pathlib import Path
+from xml.etree.ElementTree import Element
+
+from rdflib import RDF, RDFS, XSD, Literal, Namespace, URIRef
+
+from cognite.neat.constants import DEFAULT_NAMESPACE
+from cognite.neat.graph.extractors._base import BaseExtractor
+from cognite.neat.graph.models import Triple
+from cognite.neat.utils.utils import as_neat_compliant_uri
+from cognite.neat.utils.xml import get_children, iterate_tree
+
+DEXPI = Namespace("http://sandbox.dexpi.org/rdl/")
+
+
+class DexpiExtractor(BaseExtractor):
+    """
+    DEXPI-XML extractor of RDF triples
+
+    Args:
+        filepath: File path to DEXPI XML file.
+        namespace: Optional custom namespace to use for extracted triples that define data
+                    model instances. Defaults to DEFAULT_NAMESPACE.
+    """
+
+    def __init__(
+        self,
+        filepath: Path,
+        namespace: Namespace | None = None,
+    ):
+        self.filepath = filepath
+        self.namespace = namespace or DEFAULT_NAMESPACE
+        self.root = ET.parse(self.filepath).getroot()
+
+    @classmethod
+    def from_file(cls, file_path: str | Path, namespace: Namespace | None = None):
+        return cls(Path(file_path), namespace)
```

Comment:
I suggest the following:
```suggestion
    def __init__(
        self,
        filepath: Path,
        namespace: Namespace | None = None,
    ):
        self.filepath = filepath
        self.namespace = namespace or DEFAULT_NAMESPACE
        self.root = ET.parse(self.filepath).getroot()

    @classmethod
    def from_file(cls, file_path: str | Path, namespace: Namespace | None = None):
        return cls(Path(file_path), namespace)
```
```suggestion
    def __init__(
        self,
        root: Element,
        namespace: Namespace | None = None,
    ):
        self.root = root
        self.namespace = namespace or DEFAULT_NAMESPACE

    @classmethod
    def from_file(cls, file_path: str | Path, namespace: Namespace | None = None):
        return cls(ET.parse(file_path).getroot(), namespace)
```

As far as it does not hurt performance I think extractors should take the core data object in the `__init__` method that it works on the `.extract` method. 

Then, if we want to extend it in the future to for example read a dexpi from an URL it is easy to extend it with
```python
    @classmethod
    def from_url(cls, url: str, namespace: Namespace | None = None):
        import requests
        from io import BytesIO

        response = requests.get(url)
        response.raise_for_status()
        return cls(ET.parse(BytesIO(response.content)).getroot(), namespace)
```
With the current implementation the above would not be possible and you would have to dump a file to disk and read it again.

Reply:
I like this!